awx/main/tasks/helpers.py

```diff
@@ -0,0 +1,8 @@
+from django.utils.timezone import now
+from rest_framework.fields import DateTimeField
+
+
+def is_run_threshold_reached(setting, threshold_seconds):
+    last_time = DateTimeField().to_internal_value(setting.value) if setting and setting.value else DateTimeField().to_internal_value('1970-01-01T00:00')
```

Comment:
The call to this is

```python
is_run_threshold_reached(getattr(settings, 'CLEANUP_HOST_METRICS_LAST_TS', None), getattr(settings, 'CLEANUP_HOST_METRICS_INTERVAL', 30) * 86400)
```

And that's incompatible with the `setting.value` syntax, which treats this as a `Setting` object, as opposed to a datetime object.

To state my preference - I would prefer that the `awx/main` app never touches a `Setting` object directly at all, and only deals with their values.

Reply:
good catch, fixed